overlord/devicestate/handlers.go

```diff
@@ -98,13 +98,13 @@ func (m *DeviceManager) doSetModel(t *state.Task, _ *tomb.Tomb) error {
 	}
 	for snapName, snapst := range snapStates {
 		// TODO: remove this type restriction once we remodel
-		//       kernels/gadgets and add tests that ensure
+		//       gadgets and add tests that ensure
```

Comment:
I just discovered this whole loop is not unit tested

Reply:
The loop is tested in TestSetModelHandlerNewRevision - I added a kernel in there too now.